packages/odf/modals/s3-browser/delete-and-empty-bucket/DeleteBucketModal.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from 'react';
+import { CommonModalProps } from '@odf/shared/modals';
+import { S3Commands } from '@odf/shared/s3';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { LaunchModal } from '@openshift-console/dynamic-plugin-sdk/lib/app/modal-support/ModalProvider';
+import { Trans } from 'react-i18next';
+import useSWR from 'swr';
+import {
+  Modal,
+  Button,
+  Text,
+  TextInput,
+  ModalVariant,
+  ValidatedOptions,
+  TextInputTypes,
+  TextVariants,
+  TextContent,
+  FormGroup,
+  Alert,
+  Spinner,
+  AlertVariant,
+  AlertActionLink,
+} from '@patternfly/react-core';
+import { EmptyBucketResponse, getTextInputLabel } from './EmptyBucketModal';
+import { LazyEmptyBucketModal } from './lazy-delete-and-empty-bucket';
+
+type DeleteBucketModalProps = {
+  bucketName: string;
+  noobaaS3: S3Commands;
+  launcher: LaunchModal;
+  refreshTokens?: () => void;
+  setEmptyBucketResponse: React.Dispatch<
+    React.SetStateAction<EmptyBucketResponse>
+  >;
+};
+
+const DeleteBucketModal: React.FC<CommonModalProps<DeleteBucketModalProps>> = ({
+  closeModal,
+  isOpen,
+  extraProps: {
+    bucketName,
+    noobaaS3,
+    launcher,
+    refreshTokens,
+    setEmptyBucketResponse,
+  },
+}) => {
+  const { t } = useCustomTranslation();
+  const [inputValue, setInputValue] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const checkBucketContents = async () => {
+    const objects = await noobaaS3.listObjectVersions({
+      Bucket: bucketName,
+      MaxKeys: 1,
+    });
+    return objects;
+  };
+
+  const {
+    data,
+    error,
+    isLoading: isChecking,
+  } = useSWR(
+    isOpen ? ['checkBucketContents', bucketName] : null,
+    checkBucketContents
+  );
```

Comment:
you don't have to worry about async/await etc... `useSWR` will take care automatically...
```suggestion
  const {
    data,
    error,
    isLoading: isChecking,
  } = useSWR(
    `${LIST_VERSIONED_OBJECTS}/${bucketName}`,
    () => noobaaS3.listObjectVersions({
      Bucket: bucketName,
      MaxKeys: 1,
    })
  );
```